WordPressAuthenticator/GoogleSignIn/String+Random.swift

```diff
@@ -0,0 +1,12 @@
+extension String {
+
+    static func randomString(using characters: Set<Character>, withLength length: Int) -> String {
+        let allowedCharactersCount = UInt32(characters.count)
+
+        return (0..<length).reduce("") { accumulator, _ in
+           let randomOffset = Int(arc4random_uniform(allowedCharactersCount))
+           let randomCharacter = characters[characters.index(characters.startIndex, offsetBy: randomOffset)]
+           return accumulator + String(randomCharacter)
+        }
```

Comment:
Nit, but we can probably golf away `arc4random_uniform` in favour of the Swift standard library _if this doesn't need to be cryptographically secure_. If it _does need_ to be cryptographically secure (which I think is the case because it's called in the PKCE implementation) , I wonder if we should name the method to `randomSecureString` or similar?

Assuming it _does_ need to be cryptographically secure, WDYT about using `SecRandomCopyBytes` to do this on O(1) time instead of O(n)?

```suggestion
       (0..<length).reduce("") { accumulator, _ in
           let randomOffset = Int.random(in: 0...characters.count)
           let randomCharacter = characters[characters.index(characters.startIndex, offsetBy: randomOffset)]
           return accumulator + String(randomCharacter)
        }
```

Reply:
@jkmassel 

>  If it _does need_ to be cryptographically secure (which I think is the case because it's called in the PKCE implementation)

This is what the [spec](https://www.rfc-editor.org/rfc/rfc7636#section-4.1) says:

> code_verifier = high-entropy cryptographic random STRING using the unreserved characters [A-Z] / [a-z] / [0-9] / "-" / "." / "_" / "~"  from [Section 2.3 of [RFC3986]](https://www.rfc-editor.org/rfc/rfc3986#section-2.3), with a minimum length of 43 characters and a maximum length of 128 characters.

"cryptographic random STRING" makes me think it should be cryptographically secure.

Reply:
I rewrote to use `SecRandomCopyBytes` in 4629e575. However, I'm not sure how to from O(n) to O(1) without losing the part of code that picks characters from the allowed set.

Reply:
Looking over the [spec](https://www.rfc-editor.org/rfc/rfc7636#section-4.1) once more, I found this

> It is RECOMMENDED that the output of a suitable random number generator be used to create a 32-octet sequence.  The octet sequence is then base64url-encoded to produce a 43-octet URL safe string to use as the code verifier.

Adopting this approach would remove a lot of custom code on our end. I'll try it in a dedicated PR stacked on this one.

_(RTFM Gio...)_